apps/dashboard/app/models/batch_connect/session.rb

```diff
@@ -556,11 +556,26 @@ def connection_in_info?
     # Whether to allow SSH to node running the job
     # @return [Boolean] whether to allow SSH to node running the job
     def ssh_to_compute_node?
-      if !cluster_ssh_to_compute_node?.nil?
-        return cluster_ssh_to_compute_node?
-      else
-        return global_ssh_to_compute_node?
-      end
+      return cluster_and_app_ssh_to_compute_node? if override_global_ssh_to_compute_node?
+      global_ssh_to_compute_node?
+    end
+
+    def cluster_and_app_ssh_to_compute_node?
```

Comment:
We may be able to get by with just `cluster_and_app_ssh?` implying that it's to a compute node.  I'm happy to imply that it's sshing to a compute node, where else would a session shell to?